demos/text_detection_demo/src/text_recognition.cpp

```diff
@@ -53,3 +75,81 @@ std::string CTCGreedyDecoder(const std::vector<float> &data, const std::string&
     }
     return res;
 }
+
+std::string CTCBeamSearchDecoder(const std::vector<float> &data, const std::string& alphabet, char pad_symbol, double *conf, int bandwidth) {
+    std::string res = "";
+
+    const int num_classes = alphabet.length();
+
+    std::vector<BeamElement> curr;
+    std::vector<BeamElement> last;
+
+    last.push_back(BeamElement{std::vector<int>(), 1.f, 0.f});
+
+    for (std::vector<float>::const_iterator it = data.begin(); it != data.end(); it += num_classes) {
+        curr.clear();
+
+        std::vector<float> prob = softmax(it, it + num_classes);
+
+        for(const auto& candidate: last) {
+            float prob_not_blank = 0.f;
+            const std::vector<int>& candidate_sentence = candidate.sentence;
+            if (!candidate_sentence.empty()) {
+                int n = candidate_sentence.back();
+                prob_not_blank = candidate.prob_not_blank * prob[n];
+            }
+            float prob_blank = candidate.prob() * prob[(num_classes - 1)];
```

Comment:
```suggestion
            float prob_blank = candidate.prob() * prob[num_classes - 1];
```

Redundant parentheses.